Source/CarthageKit/Xcode.swift

```diff
@@ -1079,88 +1079,164 @@ public typealias BuildSchemeProducer = SignalProducer<TaskEvent<(ProjectLocator,
-					return (pathComponent as NSString).stringByAppendingPathComponent(trailingPath)
-				}
+/// Creates symlink between the dependency checkouts and the root checkouts
+///
+/// Returns a signal indicating success
+private func symlinkCheckoutPathForDependencyProject(dependency: ProjectIdentifier, subDependencies: Set<ProjectIdentifier>, rootDirectoryURL: NSURL) -> SignalProducer<(), CarthageError> {
+	let rootCheckoutsURL = rootDirectoryURL.URLByAppendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).URLByResolvingSymlinksInPath!
+	let rawDependencyURL = rootDirectoryURL.URLByAppendingPathComponent(dependency.relativePath, isDirectory: true)
+	let dependencyURL = rawDependencyURL.URLByResolvingSymlinksInPath!
+	let dependencyCheckoutsURL = dependencyURL.URLByAppendingPathComponent(CarthageProjectCheckoutsPath, isDirectory: true).URLByResolvingSymlinksInPath!
+	let subDependencyNames = subDependencies.map { $0.name }
+	let fileManager = NSFileManager.defaultManager()
 
-				do {
-					try NSFileManager.defaultManager().createSymbolicLinkAtPath(dependencyBinariesURL.path!, withDestinationPath: linkDestinationPath)
-				} catch let error as NSError {
-					return .Failure(.WriteFailed(dependencyBinariesURL, error))
+	return SignalProducer<(), CarthageError>
+		.attempt {
+			do {
+				try fileManager.createDirectoryAtURL(dependencyCheckoutsURL, withIntermediateDirectories: true, attributes: nil)
+			} catch let error as NSError {
+				if !(error.domain == NSCocoaErrorDomain && error.code == NSFileWriteFileExistsError) {
+					return .Failure(.WriteFailed(dependencyCheckoutsURL, error))
 				}
 			}
-
-			return .Success(schemeProducers)
+			return .Success()
 		}
-		.flatMap(.Merge) { schemeProducers -> SignalProducer<BuildSchemeProducer, CarthageError> in
-			return schemeProducers
-				.mapError { error in
-					switch (dependency, error) {
-					case let (_, .NoSharedFrameworkSchemes(_, platforms)):
-						return .NoSharedFrameworkSchemes(dependency, platforms)
-
-					case let (.GitHub(repo), .NoSharedSchemes(project, _)):
-						return .NoSharedSchemes(project, repo)
+		.flatMap(.Merge) { _ -> SignalProducer<(), CarthageError> in
+
+			return fileManager.carthage_enumeratorAtURL(rootCheckoutsURL, includingPropertiesForKeys: [NSURLIsDirectoryKey], options: [.SkipsSubdirectoryDescendants, .SkipsHiddenFiles])
+				.attemptMap { _, checkoutURL -> Result<(), CarthageError> in
+					let isDirectory: Bool
+					do {
+						var value: AnyObject?
+						try checkoutURL.getResourceValue(&value, forKey: NSURLIsDirectoryKey)
+						if let value = value {
+							isDirectory = value.boolValue
+						} else {
+							return .Failure(.ReadFailed(checkoutURL, nil))
+						}
+					} catch let error as NSError {
+						return .Failure(.ReadFailed(checkoutURL, error))
+					}
 
-					default:
-						return error
+					let dependencyName = checkoutURL.lastPathComponent!
+					let hasDependency = subDependencyNames.contains(dependencyName)
```

Comment:
This seems backwards. Rather than enumerating `/Checkouts/` and skipping dependencies we don't want, we should enumerate the dependencies we do want and look for their directories.